bindings/python/mata.pyx

```diff
@@ -450,7 +662,67 @@ cdef class Nfa:
         mata.intersection(
             result.thisptr, dereference(lhs.thisptr), dereference(rhs.thisptr), &product_map
         )
-        return result, {tuple(sorted(k)): v for k, v in product_map}
+        return result, {tuple(k): v for k, v in product_map}
+
+    @classmethod
+    def intersection_epsilon_preserving(cls, Nfa lhs, Nfa rhs, Symbol epsilon):
```

Comment:
Suggestion: rename to `intersection_preserving_epsilon`

Reply:
This name is clearly unfinished. The full name of this operation is `intersection_preserving_epsilon_transitions()`. I would even consider this full name. Otherwise, I would shorten the full name to something as `intersection_pres_eps_trans()`.

Your suggestion is acceptable, too. However, I feel that the word _transitions_ (shortened or not) should appear in the name to explicitly state that the epsilon **transitions** are being preserved.

Reply:
The long name is OK. I for one hate usage of the shortened phrases, currently every IDE has autocompletion, so I find it contraproductive to shortened too aggresively.

Reply:
I completely agree. `intersection_preserving_epsilon_transitions()` it is.